### PR TITLE
DBZ-2860 Explicitly declare to Quarkus that ORM XML mapping is required for the outbox extension

### PR DESCRIPTION
**Pull request by yrodiere**
https://issues.redhat.com/browse/DBZ-2860

Creating as a draft, since Quarkus 1.11.0.Beta1 hasn't been released yet.

To test this:

* Get [that PR](https://github.com/quarkusio/quarkus/pull/13929) in your local clone of Quarkus, then build it with `./mvnw -U -Dquickly`
* Go to your local clone of Debezium and get this PR
* `mvn clean install -DskipTests -DskipITs -Dquarkus.version=999-SNAPSHOT`
* `cd debezium-quarkus-outbox`
* `TESTCONTAINERS_RYUK_DISABLED=true mvn clean install  -Dquarkus.version=999-SNAPSHOT`

### CODE REVIEW

File: debezium-quarkus-outbox/deployment/src/main/java/io/debezium/outbox/quarkus/deployment/OutboxProcessor.java [hunk 1]
```diff
@@ -122,6 +127,14 @@ public void produceOutboxBuildItem(CombinedIndexBuildItem index,
         LOGGER.infof("Binding Payload as '%s'.", payloadType.name().toString());
 
         outboxEventEntityProducer.produce(new OutboxEventEntityBuildItem(aggregateIdType, payloadType));
+
+        // We enable outbox events in *all* persistence units.
```
Comment (Sanne) [inline debezium-quarkus-outbox/deployment/src/main/java/io/debezium/outbox/quarkus/deployment/OutboxProcessor.java:131]: :+1: 

For context, I don't think it's worth trying to optimise this to enable `setXmlMappingRequired(true)` more selectively: the important matter is to not load the XML parsers at all when noone is needing them.

But if a single PU needs it, might as well keep this on for all, as classes will be loaded anyway.

**Comment by gunnarmorling** [comment]
Hey @Naros, could you check this one out? Might even have it for tomorrow's 1.4.0.CR1?

**Comment by yrodiere** [comment]
> Might even have it for tomorrow's 1.4.0.CR1?

Quarkus 1.11.0.Beta1 probably won't be available by then, unless you coordinate very tightly with @gsmet .

**Comment by gsmet** [comment]
I expect the 1.11.0.Beta1 artifacts to be available sometime before 2 PM but Sonatype will decide :).

**Comment by gunnarmorling** [comment]
Ah, yeah, sorry, forgot we need that first.

**Comment by gunnarmorling** [comment]
No pressure then, we can do it for 1.4.0.Final as discussed.

**Comment by gunnarmorling** [comment]
I think this can be un-drafted by now?

**Comment by yrodiere** [comment]
@gunnarmorling Unfortunately no, the Quarkus patch couldn't make it into Beta1. It will be in Beta2, which I think will be released next week.

**Comment by gsmet** [comment]
@gunnarmorling @Naros do you have another Debezium CR planned?

If so I could release a Beta2 to unblock this thing.

**Comment by gunnarmorling** [comment]
> @gunnarmorling @Naros do you have another Debezium CR planned?

No, we don't have plans for another CR, unless any significant issues show up. Plan still is to do 1.4.0.Final in the first week of January.

**Comment by gsmet** [comment]
FYI, Quarkus 1.11.0.Beta2 has been released so this can be marked ready for review (and CI restarted).

**Comment by yrodiere** [comment]
Rebased on master just now.

**Comment by gsmet** [comment]
@Naros @gunnarmorling just a quick comment to make sure this one is included in your release. We will need it for Quarkus 1.11.

**Review by Naros — APPROVED** [review]
LGTM.

**Comment by Naros** [comment]
Thanks @gsmet and @yrodiere!

**Comment by Sanne** [comment]
Thanks all!

**Pull request merged**